File: videov2/src/main/java/com/owoh/video/activity/CameraPreviewActivity.java
package com.owoh.video.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Window;
import android.view.WindowManager;

import com.owoh.R;
import com.owoh.video.fragment.CameraPreviewFragment;
import com.tencent.bugly.crashreport.CrashReport;

/**
 * 相机预览页面
 */
public class CameraPreviewActivity extends AppCompatActivity  {

    private static final String FRAGMENT_CAMERA = "fragment_camera";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        CrashReport.initCrashReport(getApplicationContext(), "c5db1d8f24", false);
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_camera);
        if (null == savedInstanceState) {
            CameraPreviewFragment fragment = new CameraPreviewFragment();
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.container, fragment, FRAGMENT_CAMERA)
                    .addToBackStack(FRAGMENT_CAMERA)
                    .commit();
        }

    }


    @Override
    protected void onResume() {
        super.onResume();
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
    }

    @Override
    public void onBackPressed() {
        // 判断fragment栈中的个数，如果只有一个，则表示当前只处于预览主页面点击返回状态
        int backStackEntryCount = getSupportFragmentManager().getBackStackEntryCount();
        if (backStackEntryCount > 1) {
            getSupportFragmentManager().popBackStack();
        } else if (backStackEntryCount == 1) {
            CameraPreviewFragment fragment = (CameraPreviewFragment) getSupportFragmentManager()
                    .findFragmentByTag(FRAGMENT_CAMERA);
            if (fragment != null) {
                if (!fragment.onBackPressed()) {
                    finish();
                }
            }
        } else {
            super.onBackPressed();
        }
    }


}

File: videov2/src/main/java/com/owoh/video/activity/CameraPreviewActivity.kt
package com.owoh.video.activity

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Window
import android.view.WindowManager
import com.owoh.R
import com.owoh.databinding.ActivityCameraBinding
import com.owoh.video.fragment.CameraPreviewFragment

/**
 * 相机预览页面
 */
class CameraPreviewActivity : AppCompatActivity() {
        private lateinit var binding: ActivityCameraBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_camera)
        if (null == savedInstanceState) {
            val fragment = CameraPreviewFragment()
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.container, fragment, FRAGMENT_CAMERA)
                .addToBackStack(FRAGMENT_CAMERA)
                .commit()
        }

    }


    override fun onResume() {
        super.onResume()

    }

    override fun onBackPressed() {
        // 判断fragment栈中的个数，如果只有一个，则表示当前只处于预览主页面点击返回状态
        val backStackEntryCount = supportFragmentManager.backStackEntryCount
        if (backStackEntryCount > 1) {
            supportFragmentManager.popBackStack()
        } else if (backStackEntryCount == 1) {
            val fragment = supportFragmentManager
                .findFragmentByTag(FRAGMENT_CAMERA) as CameraPreviewFragment?
            if (fragment != null) {
                if (!fragment.onBackPressed()) {
                    finish()
                }
            }
        } else {
            super.onBackPressed()
        }
    }

    companion object {

        private const val FRAGMENT_CAMERA = "fragment_camera"
    }


}
